src/app.rb

```diff
@@ -0,0 +1,121 @@
+require_relative './book'
+require_relative './rental'
+require_relative './student'
+require_relative './teacher'
+require_relative './classroom'
+require_relative './person'
+
+class App
+  # setters and getters
+  attr_accessor :books, :rentals, :people
+
+  # constructor method
+  def initialize
+    @books = []
+    @rentals = []
+    @people = []
+  end
+
+  # method for listing all books.
+  def list_all_books
+    if @books.empty?
+      puts 'No books available yet!, go ahead and create one'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index + 1} Title: #{book.title}, Author: #{book.author}"
+      end
+    end
+  end
+
+  # method for listing all people
+  def list_all_people
+    if @people.empty?
+      puts 'No one has been listed yet!, kindly add a person'
+    else
+      @people.each_with_index do |person, index|
+        puts "#{index + 1} name: #{person.name} age: #{person.age} person_id: #{person.id}"
+      end
+    end
+  end
+
+  # Create a person (teacher or student, not a plain Person).
+  def create_person()
+    puts 'Want to create a student(1) or a teacher(2)? [Input number]: '
+    opt = gets.chomp
+    case opt
+    when '1'
+      puts 'Age:'
+      age = gets.chomp
+      puts 'Name:'
+      name = gets.chomp
+      puts 'Has parental_permission? [Y/N]:'
+      parental_permission = gets.chomp
+      puts 'Classroom:'
+      classroom = gets.chomp
+      new_student = Student.new(age, name, parental_permission, classroom)
```

Comment:
- [ ] There are also other issues like misarrangement of arguments and so on, and inability to pass arguments correctly for Student creation. You need to change from `parental_permission` to `parent_permission` to match the parameter name. You also need to check if parent_permission is `true` or `false` after getting the input.

  ```
  puts 'Has parental_permission? [Y/N]:'
  parent_permission = gets.chomp
  if (parent_permission == 'y') || (parent_permission == 'Y')
    parent_permission = true
  else
    parent_permission = false
  end
  puts 'Classroom:'
  classroom = gets.chomp
  new_student = Student.new(age, classroom, name, parent_permission: parent_permission)
  ```

Reply:
Change implemented